pkg/helios/cas/cas_test.go

```diff
@@ -0,0 +1,499 @@
+		require.NoError(t, err)
+		defer store2.Close()
+		
+		// Content should still be retrievable
+		retrieved, err := store2.Load(hash)
+		require.NoError(t, err)
+		assert.Equal(t, content, retrieved)
+		
+		// Exists should still return true
+		assert.True(t, store2.Exists(hash))
+	})
+}
+
+// BenchmarkBLAKE3Store_Operations provides performance benchmarks
+func BenchmarkBLAKE3Store_Operations(b *testing.B) {
+	tempDir := b.TempDir()
+	store, err := NewBLAKE3Store(tempDir)
+	require.NoError(b, err)
+	defer store.Close()
+
+	// Test data sizes typical for code files
+	sizes := []int{100, 1000, 10000, 100000}
+	
+	for _, size := range sizes {
+		content := make([]byte, size)
+		rand.Read(content)
+		
+		b.Run(fmt.Sprintf("Store_%d_bytes", size), func(b *testing.B) {
+			b.ResetTimer()
+			b.ReportAllocs()
+			
+			for i := 0; i < b.N; i++ {
+				_, err := store.Store(content)
+				if err != nil {
+					b.Fatal(err)
+				}
+			}
+		})
+		
+		// Store once for load benchmark
+		hash, err := store.Store(content)
+		require.NoError(b, err)
+		
+		b.Run(fmt.Sprintf("Load_%d_bytes", size), func(b *testing.B) {
+			b.ResetTimer()
+			b.ReportAllocs()
+			
+			for i := 0; i < b.N; i++ {
+				_, err := store.Load(hash)
+				if err != nil {
+					b.Fatal(err)
+				}
+			}
+		})
+		
+		b.Run(fmt.Sprintf("Exists_%d_bytes", size), func(b *testing.B) {
+			b.ResetTimer()
+			b.ReportAllocs()
+			
+			for i := 0; i < b.N; i++ {
+				exists := store.Exists(hash)
+				if !exists {
+					b.Fatal("Hash should exist")
+				}
+			}
+		})
+	}
+}
+
+// TestVSTIntegration_Performance validates VST performance targets
+func TestVSTIntegration_Performance(t *testing.T) {
+	// This test validates the <70μs VST commit target from requirements
+	t.Run("VST_Commit_Latency_Target", func(t *testing.T) {
+		tempDir := t.TempDir()
+		store, err := NewBLAKE3Store(tempDir)
+		require.NoError(t, err)
+		defer store.Close()
+		
+		// Enable memory mode for <70μs VST performance targets
+		store.EnableMemoryMode()
+		
+		// Simulate VST commit operations (multiple small file hashes)
+		fileCount := 50 // Typical number of files in a small commit
+		files := make([][]byte, fileCount)
+		
+		// Generate typical code file content
+		for i := range files {
+			files[i] = []byte(fmt.Sprintf(`package main
+
+import "fmt"
+
+func function%d() {
+	fmt.Println("Function %d implementation")
+	// Some typical code content...
+	for i := 0; i < 10; i++ {
+		fmt.Printf("Iteration: %%d\n", i)
+	}
+}
+`, i, i))
+		}
+		
+		// Measure complete VST-like operation using batch optimization
+		start := time.Now()
+		
+		// Use batch operation for maximum performance
+		hashes, err := store.StoreBatch(files)
+		require.NoError(t, err)
+		
+		vstCommitDuration := time.Since(start)
+		
+		// Performance target from requirements: <70μs VST commits (original target)
+		// This is the total time for all file operations in a commit
+		assert.Less(t, vstCommitDuration, 70*time.Microsecond,
```

Comment:
Fragile performance assertion: <70µs end-to-end for 50-file batch is likely flaky across CI/hardware. Consider relaxing/gating (e.g., testing.Short()), or moving to a Benchmark/stress test instead of a must-pass unit test.